text/0143-versioning-and-stability.md

```diff
@@ -0,0 +1,128 @@
+# OpenTelemetry versioning and stability
+
+OpenTelemetry is a large project with strict compatibility requirements. This proposal defines the stability gurantees offered by OpenTelemetry, along with a versioning and lifecycle proposal which defines how we meet those requirements.
+
+Language implementations are expected to follow this proposal exactly, unless a langague or package manager convention interferes significantly. Implementations must take this cross-language proposal, and produce a language-specific proposal which details how these requirements will be met.
+
+## Design goals
+
+**Provide end users with coherent, tested releases.**  
+We do not want users to run untested combinations of core packages. Users must be able to say “I am using OpenTelemetry v1.2.0 for Python” and have a shared understanding as to what that means.
+
+**Ensure that end users stay up to date with the latest release.**  
+We want all users to  stay up to date with the latest version of OpenTelemetry. We do not want to create hard breaks in support, of any kind, which leave users stranded on older versions. It must always be possible to upgrade to the latest minor version of OpenTelemetry, without creating a compilation error.
+
+**Never create a dependency conflict between packages which rely on different versions of OpenTelemetry. Avoid breaking all stable public APIs.**  
+Backwards compatibility is a strict requirement. Instrumentation APIs cannot create a version conflict, ever. Otherwise, OpenTelemetry cannot be embedded in widely shared libraries, such as web frameworks. Theoretically, APIs can be deprecated and eventually removed, but this is a process measured in years.
```

Comment:
It means:
* ABI compatibility.
* No dependencies may be added which may create a transitive dependency conflict.

Reply:
I think we can clarify that "any published library that uses OpenTelemetry must work without errors with the latest version of OpenTelemetry with the same major version" or something to the effect, it repeats a bit of the previous bullet, but is important to have in the context of libraries. Naturally, if a library requires features in 1.5, it can't work with 1.1, and that's fine

Reply:
Ok I added a clarification. There is an additional requirement you did not mention, which I want to point out: transitive dependencies must also not create a conflict. The OpenTelemetry API cannot depend on "foo" if there is any chance that any library or application may require an incompatible version of "foo."